### PR TITLE
[bug] fix n_jobs support coverage to indirectly-supported oneDAL methods

### PR DESCRIPTION
**Pull request by icfaust**
### Description

This adds a trace coverage check for all callable methods for all estimators to see if n_jobs is being properly set for oneDAL offload circumstances.  Seems that I missed a lot of them in "score" method support.  This PR is an extension on #2048 using the same testing fixtures.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: sklearnex/tests/test_common.py [hunk 1]
```diff
@@ -52,3 +76,162 @@ def test_target_offload_ban():
...
+
+
+@pytest.mark.parametrize("design_pattern", DESIGN_RULES)
+@pytest.mark.parametrize(
+    "estimator, method",
+    gen_models_info({**PATCHED_MODELS, **SPECIAL_INSTANCES}, fit=True, daal4py=False),
+)
+def test_estimator(estimator, method, design_pattern, estimator_trace):
+    # These tests only apply to sklearnex estimators
+    design_pattern(estimator_trace, estimator, method)
```
Comment (samir-nasibli) [inline sklearnex/tests/test_common.py:237]: 👍 I just love this initiative

**Review by samir-nasibli — APPROVED** [review]
Thank you @icfaust 
I think checking estimators design by the rule is good, but again disputable in case should they be in the package it self or just in the repo.

File: daal4py/sklearn/_n_jobs_support.py [hunk 2]
```diff
@@ -117,20 +117,22 @@ def method_wrapper(self, *args, **kwargs):
...
             logger.debug(
                 f"{cl.__module__}.{cl.__name__}.{method.__name__}: "
                 f"setting {n_jobs} threads (previous - {old_n_threads})"
             )
             set_n_threads(n_jobs)
-        result = method(self, *args, **kwargs)
-        if n_jobs != old_n_threads:
+            return method(self, *args, **kwargs)
+        finally:
             set_n_threads(old_n_threads)
```
Comment (Alexsandruss) [inline daal4py/sklearn/_n_jobs_support.py:133]: What is the purpose of try-finally block? Is it to guarantee reset to `old_n_threads`? What will happen if `method` throws some error? I guess there will be no traceback

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

**Review by Alexsandruss — APPROVED** [review]
Assuming intelci will finally start and pass.
